File: top150/ArrayString/q28.java
package top150.ArrayString;

public class q28 {
    public int strStr(String haystack, String needle) {
        // return haystack.indexOf(needle);
        int ptr = 0;
        for(int i = 0; i < haystack.length(); i++) {
            if(haystack.charAt(i) == needle.charAt(ptr)) {
                ptr++;
                if(ptr == needle.length()) {
                    return i-needle.length()+1;
                }
            } else {
                i -= ptr;
                ptr = 0;
            }
        }

        return -1;
    }
}

File: top150/_01_ArrayString/q28.py
class Solution:
    def strStr(self, haystack, needle):
        m = len(haystack)
        n = len(needle)
        l = 0
        while l < m-n+1:
            if haystack[l:l+n] == needle:
                return l
            else:
                l += 1
        return -1
